tests/dragonfly/cluster_mgr_test.py

```diff
@@ -0,0 +1,162 @@
+import subprocess
+import redis
+from redis import asyncio as aioredis
+from .utility import *
+from . import dfly_args
+
+BASE_PORT = 40001
+
+
+async def insert_cluster_data(cluster_client: redis.RedisCluster):
+    for i in range(1_000):
+        await cluster_client.set(i, i)
+
+
+async def check_cluster_data(cluster_client: redis.RedisCluster):
+    for i in range(1_000):
+        assert await cluster_client.get(i) == str(i)
+
+
+def run_cluster_mgr(args):
+    result = subprocess.run(["../tools/cluster_mgr.py", *args])
+    return result.returncode == 0
+
+
+@dfly_args({"proactor_threads": 2, "cluster_mode": "yes"})
+async def test_cluster_mgr(df_local_factory):
+    NODES = 3
+    masters = [df_local_factory.create(port=BASE_PORT + i) for i in range(NODES)]
+    replicas = [df_local_factory.create(port=BASE_PORT + 100 + i) for i in range(NODES)]
+    df_local_factory.start_all([*masters, *replicas])
+
+    # Initialize a cluster (all slots belong to node 0)
+    assert run_cluster_mgr(["--action=config_single_remote", f"--target_port={BASE_PORT}"])
+    for i in range(1, NODES):
+        assert run_cluster_mgr(
+            ["--action=attach", f"--target_port={BASE_PORT}", f"--attach_port={BASE_PORT+i}"]
+        )
+
+    # Feed the cluster with data and test that it works correctly
```

Comment:
I dont see that you set the slots before you start sending data.. right?
How is this working?